src/main/java/me/karakelley/http/handlers/QueryParametersHandler.java

```diff
@@ -0,0 +1,50 @@
+package me.karakelley.http.handlers;
+
+import me.karakelley.http.Request;
+import me.karakelley.http.Response;
+import me.karakelley.http.UriQueryParser;
+import me.karakelley.http.contentpresenter.ContentPresenter;
+import me.karakelley.http.exceptions.InvalidRequestException;
+import me.karakelley.http.responses.BadRequest;
+import me.karakelley.http.responses.Ok;
+
+import java.util.*;
+
+public class QueryParametersHandler implements Handler {
+  private final String CONTENT_TYPE = "Content-Type";
+  private final String TEXT_HTML = "text/html";
+  private ContentPresenter contentPresenter;
+
+  public QueryParametersHandler(ContentPresenter contentPresenter) {
+    this.contentPresenter = contentPresenter;
+  }
+
+  @Override
+  public Response respond(Request request) {
+    Response response = new Ok();
+    try {
+      return serveQueryResponse(response, request);
+    } catch (InvalidRequestException e) {
+      return new BadRequest();
+    }
+  }
+
+  private Response serveQueryResponse(Response response, Request request) {
+    Map<String, List<String>> query = UriQueryParser.parse(request.getPath());
+    if (!query.isEmpty()) {
+      response.setBody(contentPresenter.displayResources(createDescriptionList(query)));
+    }
+    response.setHeaders(CONTENT_TYPE, TEXT_HTML);
+    return response;
+  }
+
+  private String createDescriptionList(Map<String, List<String>> query) {
+    StringBuilder paramsTable = new StringBuilder();
+    paramsTable.append("<dl>");
+    for (Map.Entry<String, List<String>> entry : query.entrySet()) {
+      paramsTable.append("<dt>" + entry.getKey() + "</dt><dd>" + entry.getValue().get(0) + "</dd>");
```

Comment:
Why render some html here and some using the interface formerly known as `ContentPresenter`?